### PR TITLE
API-3494 Add a request-id to log entries for better correlation

### PR DESCRIPTION
**Pull request by bdgrider**
Adding a request id to all requests using the cls-rtracer library, like SAML proxy.

### CODE REVIEW

**Review by beaugrantham-va — CHANGES_REQUESTED** [review]
- [x] Update title
- [ ] Add description
- [ ] Link Jira story
- [ ] Resolve conflicts

File: oauth-proxy/package-lock.json [hunk 1]
```diff
@@ -3,6 +3,9328 @@
   "version": "1.0.0",
   "lockfileVersion": 1,
   "requires": true,
+<<<<<<< HEAD
```
Comment (beaugrantham-va) [inline oauth-proxy/package-lock.json:6]: This merge is incomplete.

File: oauth-proxy/tests/e2e.test.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ const {
 } = buildBackgroundServerModule("oauth-proxy test app");
 const { buildApp } = require("../index");
 const { encodeBasicAuthHeader } = require("../utils");
+require("cls-rtracer");
```
Comment (beaugrantham-va) [inline oauth-proxy/tests/e2e.test.js:20]: This seems to be unnecessary.

**Review by beaugrantham-va — CHANGES_REQUESTED** [review]

File: oauth-proxy/package-lock.json (hunk 1 above)
Comment (bdgrider) [reply oauth-proxy/package-lock.json:6]: Strange, I do not see this line in my dev environment and my branch says it is up to date.

Comment (beaugrantham-va) [reply oauth-proxy/package-lock.json]: Strange. I pushed a clean merged copy.

File: oauth-proxy/logger.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 const { createLogger, format, transports } = require("winston");
```
Comment (lukemarkwordtlibertyits) [inline oauth-proxy/logger.js:1]: There are two loggers that concern us, morgan and winston. Morgan logs the requests, whereas winston is the logger that logs messages such as log.inf("yada yada"). For the request_id to be useful we need to add it to both the morgan and winston loggers. This PR only adds the request_id to the morgan logger, we need to add the winston logger as well.

The `winstonMiddleware` function in the saml-proxy/src/logger.ts module will be a helpful reference.

**Review by lukemarkwordtlibertyits — CHANGES_REQUESTED** [review]

File: oauth-proxy/logger.js (hunk 3 above)
Comment (lukemarkwordtlibertyits) [reply oauth-proxy/logger.js:1]: Current winston logs look similar to below:
```
{
	"service": "oauth-proxy",
	"error": "invalid_grant",
	"message": "Could not refresh the client session with the provided refresh token invalid_grant (The refresh token is invalid or expired.)",
	"name": "OPError",
	"error_description": "The refresh token is invalid or expired.",
	"level": "error",
	"time": "2020-12-23T23:50:54.273Z"
}
```

**Review by lukemarkwordtlibertyits — APPROVED** [review]
Looks good

**Review by crolarlibertyva — APPROVED** [review]

**Comment by beaugrantham-va** [comment]
My only nitpick is the underscore being inconsistent with the other properties that all use dashes.

```
oauth-proxy_1         |   "remote-address": "::ffff:127.0.0.1",
oauth-proxy_1         |   "status-code": "200",
oauth-proxy_1         |   "content-length": "2464",
oauth-proxy_1         |   "request_id": "72403c60-4e98-11eb-8763-930cdbfbd945"
```

Given that saml-proxy has the same behavior, I'm okay with it for consistency, but did want to call it out.

**Review by beaugrantham-va — APPROVED** [review]

**Pull request merged**